R/dice.R

```diff
@@ -163,21 +172,31 @@ dice <- function(x, fm = NULL, SPC = TRUE, pctMissing = FALSE, fill = FALSE, str
   
   # safe wrapper to base::seq
   # NA in from/to (hz depths) not allowed
-  # TODO: from == to (0-thickness) horizons ??
   .seq.safe <- function(from, to, by) {
-    if(any(is.na(from)) | any(is.na(to))) {
+    if (any(is.na(from)) | any(is.na(to))) {
       stop('corrupt horizonation, consider using `strict = TRUE`', call. = FALSE)
     }
-    
+    if (from > to) {
+      toold <- to
+      to <- from
+      from <- toold
+    }
     return(
       seq(from = from, to = to, by = by)
     )
   }
   
+  # from == to (0-thickness) horizons -- these should be removed -AGB 2022/08/17
```

Comment:
How does this interact with `strict == TRUE`? I see the utility, but this is new behavior.

Reply:
This removes 0 thickness even when `strict=FALSE`, they should already be removed when `strict=TRUE`

Reply:
My thought is that its not really possible to have a 0 thickness slice--it has no "volume"--I could be off on this reasoning though.

Reply:
Agreed, that 0-thickness hz contains no useful information. We can return to this after dealing with other issues. Since this cleaning step will likely transcend `strict = TRUE`, we may want to move it "up" in the order of operations.